api/client/build.go

```diff
@@ -575,6 +577,28 @@ func rewriteDockerfileFrom(dockerfileName string, translator func(string, regist
 		}
 	}()
 
+	var signals []os.Signal
+	for _, sig := range []string{"INT", "TERM", "QUIT"} {
+		s, ok := signal.SignalMap[sig]
+		if ok {
+			signals = append(signals, s)
+		}
+	}
+	if len(signals) > 0 {
+		sigc := make(chan os.Signal, 1)
+		gosignal.Notify(sigc, signals...)
```

Comment:
I'm usually not a big fan of multiple signal handlers, it makes it hard to reason about.  Can we just change the defer above for the trustfile to always be closed regardless of a nil error or not?